src/book.py

```diff
@@ -741,11 +741,168 @@ def _read_bitpanda_pro_trades(self, file_path: Path) -> None:
+                "Asset market price currency",
+                "Asset class",
+                "Product ID",
+                "Fee",
+                "Fee asset",
+                "Spread",
+                "Spread Currency",
+            ]:
+                try:
+                    line = next(reader)
+                except StopIteration:
+                    raise RuntimeError(f"Expected header not found in file {file_path}")
+
+            for (
+                _tx_id,
+                csv_utc_time,
+                operation,
+                _inout,
+                amount_fiat,
+                fiat,
+                amount_asset,
+                asset,
+                asset_price,
+                asset_price_currency,
+                asset_class,
+                _product_id,
+                fee,
+                fee_currency,
+                _spread,
+                _spread_currency,
+            ) in reader:
+                row = reader.line_num
+
+                # make RFC3339 timestamp ISO 8601 parseable
+                if csv_utc_time[-1] == "Z":
+                    csv_utc_time = csv_utc_time[:-1] + "+00:00"
+
+                # timezone information is already taken care of with this
+                utc_time = datetime.datetime.fromisoformat(csv_utc_time)
+
+                # transfer ops seem to be akin to airdrops. In my case I got a
+                # CocaCola transfer, which I don't want to track. Would need to
+                # be implemented if need be.
+                if operation == "transfer":
+                    log.warning(
+                        f"'Transfer' operations are not "
+                        f"implemented, skipping row {row} of file {file_path}"
+                    )
+                    continue
+
+                # fail for unknown ops
+                try:
+                    operation = operation_mapping[operation]
+                except KeyError:
+                    raise RuntimeError(f"Unsupported operation '{operation}'")
+
+                if operation in ["Deposit", "Withdrawal"]:
+                    if asset_class == "Fiat":
+                        change = misc.force_decimal(amount_fiat)
+                        if fiat != asset:
+                            raise RuntimeError(
+                                f"Asset {asset} should be {fiat} in "
+                                f"row {row} of file {file_path}"
+                            )
+                    elif asset_class == "Cryptocurrency":
+                        change = misc.force_decimal(amount_asset)
+                    else:
+                        raise RuntimeError(
+                            f"Unknown asset class {asset_class}: Should be 'Fiat' or "
+                            f"'Cryptocurrency' in row {row} of file {file_path}"
+                        )
+                elif operation in ["Buy", "Sell"]:
+                    if asset_price_currency != config.FIAT:
```

Comment:
You mixed up `config.FIAT` with Euro (in the error message, in `set_price_db`, and when adding Buy/sell operations).
I think it should be fine, if you replace all `Euro`/`EUR` with `upper(config.FIAT)`.